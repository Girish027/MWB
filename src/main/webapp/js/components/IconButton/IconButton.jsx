import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Button } from '@tfs/ui-components';
import { colors } from '../../styles';

export default class IconButton extends Component {
  constructor(props) {
    super(props);
    this.props = props;
    this.onFocus = this.onFocus.bind(this);
    this.clearFocus = this.clearFocus.bind(this);

    this.state = {
      focus: false,
    };
  }

  onFocus() {
    this.setState({
      focus: true,
    });
  }

  getFillColor = () => {
    const {
      focusedColor,
      defaultColor,
      disabledColor,
      disabled,
    } = this.props;
    const { focus } = this.state;

    if (disabled) {
      return disabledColor;
    }
    return focus ? focusedColor : defaultColor;
  }

  getStrokeColor = () => {
    const {
      strokeColor,
    } = this.props;
    return strokeColor;
  }

  clearFocus() {
    this.setState({
      focus: false,
    });
  }

  render() {
    const {
      onClick,
      icon: Icon,
      height,
      width,
      padding,
      title,
      disabled,
      className,
      styleOverride,
    } = this.props;

    return (
      <Button
        onClick={onClick}
        type="flat"
        styleOverride={{
          height: 'auto',
          minHeight: '0px',
          border: 'none',
          padding,
          backgroundColor: 'initial',
          color: '#004C97',
          ':hover': {
            color: '#003467',
            boxShadow: 'none',
          },
          ':focus': {
            outline: 'none',
          },
          ...styleOverride,
        }}
        disabled={disabled}
        onBlur={this.clearFocus}
        onFocus={this.onFocus}
        onMouseEnter={this.onFocus}
        onMouseOver={this.onFocus}
        onMouseLeave={this.clearFocus}
        onMouseOut={this.clearFocus}
        name="action-button"
        title={title}
        tabIndex={0}
      >
        <Icon
          fill={this.getFillColor()}
          stroke={this.getStrokeColor()}
          width={width}
          height={height}
        />
      </Button>
    );
  }
}

IconButton.propTypes = {
  icon: PropTypes.func.isRequired,
  onClick: PropTypes.func,
  width: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  height: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  padding: PropTypes.string,
  focusedColor: PropTypes.string,
  defaultColor: PropTypes.string,
  disabledColor: PropTypes.string,
  title: PropTypes.string,
  disabled: PropTypes.bool,
  styleOverride: PropTypes.object,
};

IconButton.defaultProps = {
  onClick: () => {},
  width: 10,
  height: 10,
  padding: '1px',
  focusedColor: colors.prussianBlue,
  defaultColor: colors.cobalt,
  strokeColor: 'none',
  disabledColor: colors.disabledText,
  title: '',
  disabled: false,
  styleOverride: {},
};
