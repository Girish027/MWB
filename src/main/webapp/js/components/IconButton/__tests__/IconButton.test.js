import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import IconButton from 'components/IconButton';
import { Plus } from '@tfs/ui-components';

describe('<IconButton />', () => {
  let wrapper;

  const props = {
    onClick: jest.fn(),
    icon: Plus,
  };

  describe('Creating an instance', () => {
    test('should exist', () => {
      wrapper = shallow(<IconButton
        {...props}
      />);
      expect(wrapper.exists()).toBe(true);
    });
  });

  describe('Snapshots', () => {
    test('renders correctly when icon is not in focus', () => {
      wrapper = shallow(<IconButton
        {...props}
      />);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    test('renders correctly when icon is in focus', () => {
      wrapper = shallow(<IconButton
        {...props}
      />);
      wrapper.setState({
        focus: true,
      });
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    test('renders correctly when icon is disabled', () => {
      wrapper = shallow(<IconButton
        {...props}
        disabled
      />);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    test('renders correctly with title', () => {
      wrapper = shallow(<IconButton
        {...props}
        title="test title"
      />);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    test('renders correctly when icon with extended props', () => {
      wrapper = shallow(<IconButton
        {...props}
        width={20}
        height={20}
        strokeColor="#ffffff"
        focusedColor="red"
        defaultColor="yellow"
      />);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });

  describe('Functionality:', () => {
    beforeEach(() => {
      wrapper = shallow(<IconButton
        {...props}
      />);
    });

    describe('onFocus', () => {
      test('should set current state of focus to have value true', () => {
        expect(wrapper.state().focus).toBe(false);
        wrapper.instance().onFocus();
        expect(wrapper.state().focus).toBe(true);
      });
    });

    describe('clearFocus', () => {
      test('should set current state of focus to have value false', () => {
        wrapper.setState({
          focus: true,
        });
        wrapper.instance().clearFocus();
        expect(wrapper.state().focus).toBe(false);
      });
    });
  });
});
