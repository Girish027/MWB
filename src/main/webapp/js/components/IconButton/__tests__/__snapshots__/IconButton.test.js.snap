// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconButton /> Snapshots renders correctly when icon is disabled 1`] = `
<withTheme(Button)
  disabled={true}
  name="action-button"
  onBlur={[Function]}
  onClick={[MockFunction]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  styleOverride={
    Object {
      ":focus": Object {
        "outline": "none",
      },
      ":hover": Object {
        "boxShadow": "none",
        "color": "#003467",
      },
      "backgroundColor": "initial",
      "border": "none",
      "color": "#004C97",
      "height": "auto",
      "minHeight": "0px",
      "padding": "1px",
    }
  }
  tabIndex={0}
  title=""
  type="flat"
>
  <Memo(withTheme(Plus))
    fill="#9B9B9B"
    height={10}
    stroke="none"
    width={10}
  />
</withTheme(Button)>
`;

exports[`<IconButton /> Snapshots renders correctly when icon is in focus 1`] = `
<withTheme(Button)
  disabled={false}
  name="action-button"
  onBlur={[Function]}
  onClick={[MockFunction]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  styleOverride={
    Object {
      ":focus": Object {
        "outline": "none",
      },
      ":hover": Object {
        "boxShadow": "none",
        "color": "#003467",
      },
      "backgroundColor": "initial",
      "border": "none",
      "color": "#004C97",
      "height": "auto",
      "minHeight": "0px",
      "padding": "1px",
    }
  }
  tabIndex={0}
  title=""
  type="flat"
>
  <Memo(withTheme(Plus))
    fill="#003467"
    height={10}
    stroke="none"
    width={10}
  />
</withTheme(Button)>
`;

exports[`<IconButton /> Snapshots renders correctly when icon is not in focus 1`] = `
<withTheme(Button)
  disabled={false}
  name="action-button"
  onBlur={[Function]}
  onClick={[MockFunction]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  styleOverride={
    Object {
      ":focus": Object {
        "outline": "none",
      },
      ":hover": Object {
        "boxShadow": "none",
        "color": "#003467",
      },
      "backgroundColor": "initial",
      "border": "none",
      "color": "#004C97",
      "height": "auto",
      "minHeight": "0px",
      "padding": "1px",
    }
  }
  tabIndex={0}
  title=""
  type="flat"
>
  <Memo(withTheme(Plus))
    fill="#004c97"
    height={10}
    stroke="none"
    width={10}
  />
</withTheme(Button)>
`;

exports[`<IconButton /> Snapshots renders correctly when icon with extended props 1`] = `
<withTheme(Button)
  disabled={false}
  name="action-button"
  onBlur={[Function]}
  onClick={[MockFunction]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  styleOverride={
    Object {
      ":focus": Object {
        "outline": "none",
      },
      ":hover": Object {
        "boxShadow": "none",
        "color": "#003467",
      },
      "backgroundColor": "initial",
      "border": "none",
      "color": "#004C97",
      "height": "auto",
      "minHeight": "0px",
      "padding": "1px",
    }
  }
  tabIndex={0}
  title=""
  type="flat"
>
  <Memo(withTheme(Plus))
    fill="yellow"
    height={20}
    stroke="#ffffff"
    width={20}
  />
</withTheme(Button)>
`;

exports[`<IconButton /> Snapshots renders correctly with title 1`] = `
<withTheme(Button)
  disabled={false}
  name="action-button"
  onBlur={[Function]}
  onClick={[MockFunction]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  styleOverride={
    Object {
      ":focus": Object {
        "outline": "none",
      },
      ":hover": Object {
        "boxShadow": "none",
        "color": "#003467",
      },
      "backgroundColor": "initial",
      "border": "none",
      "color": "#004C97",
      "height": "auto",
      "minHeight": "0px",
      "padding": "1px",
    }
  }
  tabIndex={0}
  title="test title"
  type="flat"
>
  <Memo(withTheme(Plus))
    fill="#004c97"
    height={10}
    stroke="none"
    width={10}
  />
</withTheme(Button)>
`;
